frontend/packages/core-api-client/src/validation/Recipient.ts

```diff
@@ -0,0 +1,30 @@
+import { FieldKind, FormWithRecord, Record } from '../types';
+import { getFieldKind } from '../utils';
+
+export const validateRecipientHierarch = (
+  formWithRecordList: FormWithRecord<Record>[],
+): void => {
+  formWithRecordList.forEach(({ form, record }, idx, arr) => {
+    if (form.id !== record.formId) throw new Error();
+
+    if (idx === 0) {
+      // assert no key field
+      if (form.fields.find((f) => f.key)) throw new Error();
```

Comment:
this is about checking if this is the root individual right? why shouldn't the root have a key? first and last name together could be a key. shouldn't we rather check for reference here?